cssom-view-1/Overview.bs

```diff
@@ -363,10 +363,47 @@ the following steps must be run:
-
-  Note: In the case the VisualViewport was scrolled, yielding no translation changes in the document, emit the <a event>scrollend</a> on the VisualViewport just as the <a event>scroll</a> event is emitted for VisualViewport scrolling.
 </ol>
 
+When a user agent is to <dfn for="viewport" export>perform a scroll</dfn> of a <a>viewport</a> to a given position <var>position</var> and optionally a scroll behavior <var>behavior</var>
+(which is "<code>auto</code>" if omitted) it must perform a coordinated viewport scroll by following these steps:
+
+1. Let <var>doc</var> be the <a>viewport's</a> associated {{Document}}.
+1. Let <var>vv</var> be the {{VisualViewport}} whose <a for=visualviewport>associated document</a> is <var>doc</var>.
+1. Let <var>maxX</var> be the difference between <a>viewport</a>'s <a>scrolling box</a>'s width and the value of <var>vv</var>'s <a attribute
+    for=VisualViewport lt=width>width</a> attribute.
+1. Let <var>maxY</var> be the difference between <a>viewport</a>'s <a>scrolling box</a>'s height and the value of <var>vv</var>'s <a attribute
+    for=VisualViewport lt=height>height</a> attribute.
+1. Let <var>dx</var> be the horizontal component of <var>position</var> - the value <var>vv</var>'s <a attribute for=VisualViewport lt=pageLeft>pageLeft</a> attribute
+1. Let <var>dy</var> be the vertical component of <var>position</var> - the value of <var>vv</var>'s <a attribute for=VisualViewport lt=pageTop>pageTop</a> attribute
+1. Let <var>visual x</var> be the value of <var>vv</var>'s <a attribute for=VisualViewport lt=offsetLeft>offsetLeft</a> attribute.
+1. Let <var>visual y</var> be the value of <var>vv</var>'s <a attribute for=VisualViewport lt=offsetTop>offsetTop</a> attribute.
+1. Let <var>visual dx</var> be min(<var>maxX</var>, max(0, <var>visual x</var> + <var>dx</var>)) - <var>visual x</var>.
+1. Let <var>visual dy</var> be min(<var>maxY</var>, max(0, <var>visual y</var> + <var>dy</var>)) - <var>visual y</var>.
+1. Let <var>layout dx</var> be <var>dx</var> - <var>visual dx</var>
+1. Let <var>layout dy</var> be <var>dy</var> - <var>visual dy</var>
+1. Let <var>element</var> be <var>doc</var>'s root element if there is one, null otherwise.
+1. <a for="/">Perform a scroll</a> of the <a>viewport</a>'s <a>scrolling box</a> to its current scroll position + (<var>layout dx</var>, <var>layout dy</var>) with <var>element</var> as the
+    associated element, and <var>behavior</var> as the scroll behavior.
+1. <a for="/">Perform a scroll</a> of <var>vv</var>'s <a>scrolling box</a> to its current scroll position + (<var>visual dx</var>, <var>visual dy</var>) with <var>element</var> as the associated
+    element, and <var>behavior</var> as the scroll behavior.
+
+Note: When performing a coordinated scroll, the user agent distributes the scroll between the visual and layout viewports as needed. It computes the required total delta and applies it
```

Comment:
My mental model is that the visual viewport “bumps up against” the edge of the layout viewport, pushing it around. This is effectively the same as this “distribute the scroll”, but perhaps could lead to a cleaner description of the algorithm.

Reply:
It would be totally awesome for the spec to have a nice animation showing how visual and layout viewports interact.